_posts/2017-01-09-Jak-psat-php-Zavislosti+verzovaci-systemy+debugging.md

```diff
@@ -0,0 +1,38 @@
+
+Pro základní používání Composeru si vystačíte si příkazy pro přidání, aktualizaci a odebrání balíčků a nemusíte ho dál řešit. Ovládá se pomocí příkazového řádku a souboru `/composer.json`. Ten obsahuje seznam všech balíčků a jejich dalších závislosti. Samotné soubory s kódem jsou potom standardně uloženy ve složce `/vendor`. 
+
+Jakmile začnete používat framework nebo cizí kód, Composer se stane neocenitelným společníkem. Rozhodně se vyplatí ho začít používat co nejdříve. Za zmínku stojí, že pomocí něj můžou ostatní používat jednoduše kód váš. Více o jeho instalaci a použití najdete na <https://getcomposer.org/doc/00-intro.md>. 
+
+Obzvláště v již existujícím kódu můžete narazit na starší systém pro správu PHP rozšíření [PEAR](https://pear.php.net/). Ten se používá pro správu PHP komponent, neřeší ovšem závislosti tak elegantně jako Composer a neposkytuje žádnou jinou zásadnější výhodu. Některé knihovny jsou ale historicky dostupné jen přes PEAR, v některých aplikacích se proto používá dodnes. 
+
+# Verzovací systém (Version Control System, zkráceně verzování nebo VCS)
+Jsou mezi námi dlouho a používá je míň lidí, než by bylo vhodné. Verzovací systémy slouží ke snadné správě kódu v čase. Můžete se podívat, kdy jaká změna v kódu nastala a který vývojář ji provedl. Umožňují jednoduše provést "rollback" (návrat v čase) nebo souběžný vývoj více různých větví programu a jejich následné sloučení do jednoho finálního projektu. Při vyvíjení v týmu větším než tři lidi už jsou prakticky nepostradatelné. 
+ 
+Verzovací systém ze své podstaty poskytuje nástroje a výhody zálohování. Ve zdrojových kódech nemusí být zakomentované bloky kódu, které využijeme "někdy příště"; z verzování se v případě potřeby dají snadno vytáhnout z historie. Pomocí verzovacích systému se dá i jednoduše zálohovat i jinam než na vlastní disk, o tom ale více příště. 
+   
+Nejpoužívanějším verzovacím systémem je git (pozor, neplést s <github.com>, který git používá, je to ale cloudový správce kódu>). Git vymyslel a napsal [Linus Torvalds](https://cs.wikipedia.org/wiki/Linus_Torvalds), když vyvíjel Linux. Používá se pro správu takových projektů jako jsou linuxové jádro nebo programovací jazyk Ruby. Je zpočátku složitější na pochopení, ale jeho užitečnost daleko přesahuje počáteční náklady na nastudování. Doporučuju ho používat v podstatě na všechny projekty (mnohdy i na ty neprogramátorské, kde je potřeba pamatovat si historii, kupříkladu diplomek či jiných rozsáhlých textů). Mezi dalšími verzovacími systémy se používají [Subversion](https://subversion.apache.org/) od [Apache](https://www.apache.org/) a [Mercurial](https://www.mercurial-scm.org/) - práce s nimi je podobná jako s gitem, ale žádný jsem zatím nepoužíval a tak je nemůžu doporučit ani od nich odradit. 
```

Comment:
na začátky s gitem chci napsat extra článek, tak ať se to nedubluje tak moc. Tu má být teorie, praxe až bude potřeba